module/module.go

```diff
@@ -169,14 +174,14 @@ func NewModule(ctx context.Context, address string, logger *zap.SugaredLogger) (
 }
 
 // NewModuleFromArgs directly parses the command line argument to get its address.
-func NewModuleFromArgs(ctx context.Context, logger *zap.SugaredLogger) (*Module, error) {
+func NewModuleFromArgs(ctx context.Context, logger logging.ZapCompatibleLogger) (*Module, error) {
 	if len(os.Args) < 2 {
 		return nil, errors.New("need socket path as command line argument")
 	}
 	return NewModule(ctx, os.Args[1], logger)
 }
 
-// NewLoggerFromArgs can be used to create a golog.Logger at "DebugLevel" if
+// NewLoggerFromArgs can be used to create a logging.Logger at "DebugLevel" if
 // "--log-level=debug" is the third argument in os.Args and at "InfoLevel"
 // otherwise. See config.Module.LogLevel documentation for more info on how
 // to start modules with a "log-level" commandline argument.
```

Comment:
Can't comment on the line, but I'd be in favor of having our go module utility return a `zLogger` (as `logging.Logger`) instead of a `golog.Logger`. I understand the desire to "test" the backward compatibility, but I'd rather users have the best code to build their modules off of as possible, and if we're moving away from golog, then using it in our examples seems undesirable. We could add tests (not necessarily in this PR, I don't want to make you edit this too much haha) that golog can still be used, and `edaniels/golog` could become a test only dependency (I _think_ if a user is not importing test files, they won't download `edaniels/golog`).

Reply:
Furthermore, could you explain one more time why these changes may be breaking for old go modules?

Reply:
Yeah I was torn about these examples also doubling as our backwards compatibility testing. Because we're breaking compatility, I'm going to have modules only know about `logging.Logger`. Which I think will address all of the concerns.